### ISSUE TITLE
OTM under a standards body?

### ISSUE THREAD

**Issue by trevor-vaughan** [iriusrisk/OpenThreatModel#22]
Discussions aren't active so I figure I'd start the thread here.

Are there plans to pursue OTM under one of the standards bodies?

While the standard itself seems reasonable, pushing for wider adoption is difficult when the standard is vendor-housed.

**Comment by stevespringett** [comment]
I think theres community support for OTM to be an OWASP project. Also, if IriusRisk would like OTM to be an international standard, Ecma should be seriously considered. OWASP and Ecma have built a working model that's community-based while ensuring the TC is actively involved. CycloneDX is the first to leverage the working model. I can make introductions if desired.

**Comment by stephendv1** [comment]
Yes, there is always concern when a vendor is seen to control a standard/format.  IMO, it is too early to go for a heavyweight standards body that adds too much bureaucratic overhead.  An OWASP project seems like a faster alternative given where we are with OTM currently.  There is some interest with other projects and I think it would help adoption if we had at least 2 other tools using the format.  E.g. pytm, Threat Dragon.

**Comment by trevor-vaughan** [comment]
That's certainly reasonable.

Is pytm still alive? Last I checked it seemed functional but not really progressing.

**Comment by jgadsden** [comment]
Yes, pytm is still very much alive and is referenced by other projects, @izar to update us on this maybe

**Comment by izar** [comment]
Yup, pytm is very much alive. We have a lot going on behind the scenes, and at some point, we will have a fresh update. 
Regarding OTM, pytm needs changes to it to be able to actually use it - namely, making the x/y attributes not mandatory, as pytm has no concept of a graphical representation.

Just a couple of days ago we were discussing it at ThreatModCon and many of us agree with @stevespringett - we should work towards making OTM an external standard.

**Comment by stephendv1** [comment]
@jgadsden what say you about Threat Dragon also using OTM as a supported format?
@izar x/y co-ordinates can easily be made optional.

**Comment by izar** [comment]
@stephendv1 I went looking at it this morning and either there have been changes I hadn't seen or I had misread the spec (more likely....) - the x/y are only mandatory on Diagram type of Representation, which makes perfect sense.

OTOH....how about adding P to CIA ?

**Comment by jgadsden** [comment]
Yes, I agree @stephendv1 , and I have labelled the [issue in Threat dragon](https://github.com/OWASP/threat-dragon/issues/440#issuecomment-1800041704) for version 2.2 (which is the next minor version) - although no guarantee that we can find someone to do it

**Comment by jgadsden** [comment]
@stephendv1 we have some good news in that  @stevespringett and Matthew McDonald are working on OTM being a supported format for Threat Dragon

**Comment by jgadsden** [comment]
The Threat Dragon file/JSON schema is a bit quirky, with two versions for 1.x and 2.x
If OTM becomes and open standard then Threat Dragon version 3 could use it as its file format instead of the existing incompatible versions 1.x and versions 2.x formats

**Comment by stephendv1** [comment]
That is great news! Does threat dragon need many additional changes to the
spec based on what’s published currently?

*Stephen de Vries*
CEO

+34 616 33 81 38 <+34+616+33+81+38>
@stephendv <https://twitter.com/stephendv>

[image: ContinuumSecurity] <https://www.continuumsecurity.net/>

**Comment by jgadsden** [comment]
Good point, I have raised an issue on Threat Dragon : [Use OTM as the default file format #850](https://github.com/OWASP/threat-dragon/issues/850)
and have raised an issue for OTM to identify any extensions need by OTM to cover all the information contained within Threat Dragon files : #26

**Comment by stevespringett** [comment]
Regarding "If OTM becomes an open standard...". OWASP is now a member of Ecma International.  The CycloneDX community has worked with Ecma on developing a community-based standardization process that is going to be the model of the future. It would be possible to leverage what CycloneDX and Ecma have already created and use it as a template to create their own technical committee under Ecma with the end goal of making OTM an Ecma standard. Ecma also has liaison agreements with ISO and other standards bodies, so theoretically, OTM could also be an ISO standard by way of Ecma.

If this is of interest to IriusRisk and the community, please let me know and we can discuss next steps.

**Comment by jgadsden** [comment]
certainly from my point of view this is a good way forward
Threat Dragon will be working towards full integration with OTM whatever the outcome

**Comment by jmehnle** [comment]
What is the current license of the OTM spec? As it stands, the repository specifies no license at all, which makes it fully proprietary.

**Comment by dfernandezvigo** [comment]
Thank you @jmehnle  for bringing this up! The intent is for the entire OTM specification to be under the **Creative Commons Attribution-ShareAlike 4.0 International License**, not just the README. However, we recognize that the current repository structure might cause confusion.  

To address this, we will ensure the license is explicitly clarified and added at the root of the repository to avoid ambiguity. We appreciate your feedback and will work on updating this promptly.  

If you have any further suggestions or concerns, please let us know!
